### PR TITLE
feat: Remove partial fetch

### PR DESCRIPTION
**Pull request by roneli**
Streamlined diagnostics as part of the execution, this replaces partial fetch, but essentially does the same, working with diags.
User can set resolver error classifiers instead of default ones.
Moved execution out of schema to make the layout more sensible 

TODO:
- [ ] More testing to execution edge cases (columns, classifier, failing DB insert)
- [x] Add back tests removed (resource etc')
- [x] Fix current tests
- [ ] Add diagnostics.Squash
- [x] Cleanup diags/errors from dead / unused code

### CODE REVIEW

File: provider/execution/execution.go [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+func (e TableExecutor) doMultiplexResolve(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource) (uint64, diag.Diagnostics) {
+	var (
+		clients        []schema.ClientMeta
+		diagsChan      = make(chan diag.Diagnostics)
+		totalResources uint64
+		wg             sync.WaitGroup
+	)
+	clients = e.Table.Multiplex(meta)
+	meta.Logger().Debug("multiplexing client", "count", len(clients), "table", e.Table.Name)
+	defer close(diagsChan)
```
Comment (disq) [inline provider/execution/execution.go:115]: move `close(diagsChan)` (without defer) before `wg.Wait()` to prevent deadlock.

File: provider/execution/execution.go [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+		return 0, diags.Append(FromError(err, WithResource(e.ResourceName), WithErrorClassifier))
+	}
+
+	res := make(chan interface{})
+	var resolverErr error
+	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				client.Logger().Error("table resolver recovered from panic", "table", e.Table.Name, "stack", string(debug.Stack()))
+				resolverErr = FromError(fmt.Errorf("table resolver panic: %s", r), WithResource(e.ResourceName), WithSeverity(diag.PANIC),
```
Comment (disq) [inline provider/execution/execution.go:184]: while we're at it is it possible to think about where would the stacktrace go in resolverErr? maybe a separate error type (and remove diag.PANIC)?

File: provider/execution/execution.go [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+			diags = diags.Append(err)
+			continue
+		}
+		resources = append(resources, resource)
+	}
+
+	// only top level tables should cascade
+	shouldCascade := parent == nil
+	resources, dbDiags := e.copyDataIntoDB(ctx, resources, shouldCascade)
+	diags = diags.Append(dbDiags)
```
Comment (disq) [inline provider/execution/execution.go:245]:
```suggestion
	diags = diags.Append(dbDiags...)
```

File: provider/execution/execution.go [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+	totalCount := uint64(len(resources))
+
+	// Finally, resolve relations of each resource
+	for _, rel := range e.Table.Relations {
+		meta.Logger().Debug("resolving table relation", "table", e.Table.Name, "relation", rel.Name)
+		for _, r := range resources {
+			// ignore relation resource count
+			if _, innerDiags := e.WithTable(rel).Resolve(ctx, meta, r); innerDiags.HasDiags() {
+				diags = diags.Append(innerDiags)
```
Comment (disq) [inline provider/execution/execution.go:254]:
```suggestion
				diags = diags.Append(innerDiags...)
```

File: provider/execution/execution.go [hunk 5]
```diff
@@ -0,0 +1,371 @@
...
+		partialFetchResources = append(partialFetchResources, resources[id])
+	}
+	return partialFetchResources, diags
+}
+
+func (e *TableExecutor) resolveResourceValues(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			e.Logger.Error("resolve resource recovered from panic", "table", e.Table.Name, "stack", string(debug.Stack()))
+			err = FromError(fmt.Errorf("column resolve panic: %s", r), WithResource(e.ResourceName), WithSeverity(diag.PANIC),
```
Comment (disq) [inline provider/execution/execution.go:295]: Same panic comments here, where would the stack trace go?

File: provider/provider.go [hunk 6]
```diff
@@ -3,17 +3,16 @@ package provider
...
 	"embed"
-	"errors"
 	"fmt"
 	"os"
 	"strconv"
 	"sync"
 	"sync/atomic"
 
+	"github.com/cloudquery/cq-provider-sdk/provider/execution"
+
```
Comment (disq) [inline provider/provider.go:13]: rogue space

File: provider/execution/execution.go [hunk 7]
```diff
@@ -0,0 +1,371 @@
...
+	// extraFields to be passed to each created resource in the execution
+	extraFields map[string]interface{}
+	// When the execution started
+	executionStart time.Time
+	// parent is the parent TableExecutor
+	parent *TableExecutor
+}
+
+// CreateTableExecutor creates a new TableExecutor for given schema.Table
+func CreateTableExecutor(resourceName string, db Storage, logger hclog.Logger, table *schema.Table,
```
Comment (disq) [inline provider/execution/execution.go:65]: Why not `NewTableExecutor`?

File: provider/execution/storage_test.go [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+import (
+	"context"
+	"time"
+
+	"github.com/cloudquery/cq-provider-sdk/provider/schema"
+	"github.com/jackc/pgx/v4"
+)
+
+type doNothingStorage struct {
```
Comment (disq) [inline provider/execution/storage_test.go:11]: `noopStorage`

Comment (disq) [reply provider/execution/storage_test.go:11]: also
```
var _ Storage = (*noopStorage)(nil)
```

Comment (disq) [reply provider/execution/storage_test.go:11]: But unused? then why include it

**Review by disq — CHANGES_REQUESTED** [review]
LGTM overall

File: provider/execution/storage_test.go (hunk 8 above)
Comment (roneli) [reply provider/execution/storage_test.go:11]: It is used in the execution_test.go

File: provider/execution/execution.go (hunk 7 above)
Comment (roneli) [reply provider/execution/execution.go:65]: fixed

File: provider/provider.go (hunk 6 above)
Comment (roneli) [reply provider/provider.go:13]: fixed

File: provider/execution/execution.go (hunk 5 above)
Comment (roneli) [reply provider/execution/execution.go:295]: Added stack as details

File: provider/execution/execution.go (hunk 4 above)
Comment (roneli) [reply provider/execution/execution.go:254]: the .Append function does this automatically

File: provider/execution/errors.go [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+		}
+		for _, o := range opts {
+			o(e)
+		}
+		return diag.Diagnostics{e}
+	}
+}
+
+// NewError creates an ExecutionError from given error
+func NewError(severity diag.Severity, dt diag.DiagnosticType, resource, summary string, args ...interface{}) *Error {
```
[the remark this answers is not in the document]
Comment (roneli) [reply provider/execution/errors.go:115]: done

File: provider/execution/execution.go (hunk 4 above)
Comment (disq) [reply provider/execution/execution.go:254]: oh... is `Append` a good name for it (ie. promising order) what about `.Add`?

Comment (roneli) [reply provider/execution/execution.go:239]: changed

File: provider/execution/execution.go [hunk 10]
```diff
@@ -248,7 +248,7 @@ func (e TableExecutor) resolveResources(ctx context.Context, meta schema.ClientM
 	return totalCount, diags
 }
 
-func (e TableExecutor) copyDataIntoDB(ctx context.Context, resources schema.Resources, shouldCascade bool) (schema.Resources, diag.Diagnostics) {
+func (e TableExecutor) copyToStorage(ctx context.Context, resources schema.Resources, shouldCascade bool) (schema.Resources, diag.Diagnostics) {
```
Comment (disq) [inline provider/execution/execution.go:251]: `store` ? `save`?

File: provider/diag/error.go [hunk 11]
```diff
@@ -1,7 +1,7 @@
 package diag
 
-// ExecutionError is a generic error returned when execution is run, ExecutionError satisfies
-type ExecutionError struct {
+// BaseError is a generic error returned when execution is run, ExecutionError satisfies Diagnostic interface
```
Comment (disq) [inline provider/diag/error.go:3]:
```suggestion
// BaseError is a generic error returned when execution is run, satisfies Diagnostic interface
```

File: provider/diag/error.go [hunk 12]
```diff
@@ -22,34 +22,38 @@ type ExecutionError struct {
 	diagnosticType DiagnosticType
 }
 
-func (e ExecutionError) Severity() Severity {
+// NewBaseError creates an BaseError from given error
```
Comment (disq) [inline provider/diag/error.go:25]:
```suggestion
// NewBaseError creates a BaseError from given error
```

**Review by disq — APPROVED** [review]
LGTM with nits

File: provider/execution/execution.go (hunk 2 above)
Comment (roneli) [reply provider/execution/execution.go:174]: It will basically be a copy paste more or less / syntactic sugar so not sure, can add, maybe in a later PR?

Comment (disq) [reply provider/execution/execution.go:174]: Adding it as detail was spot on, no need to do anything else I think

**Review by disq — APPROVED** [review]
